std/assembly/array.ts

```diff
@@ -472,11 +475,7 @@ export class Array<T> extends ArrayBufferView {
     if (isBoolean<T>())   return joinBooleanArray(dataStart, length, separator);
     if (isInteger<T>())   return joinIntegerArray<T>(dataStart, length, separator);
     if (isFloat<T>())     return joinFloatArray<T>(dataStart, length, separator);
-
-    if (ASC_SHRINK_LEVEL < 1) {
-      if (isString<T>())  return joinStringArray(dataStart, length, separator);
-    }
-    // For rest objects and arrays use general join routine
```

Comment:
Hmm, this conditional guard potentially could decrease code size

Reply:
Yeah, not sure about this one as well. Removed it because `joinReferenceArray` wasn't compatible with strings anymore, but there's probably something better we can do.